### DISCUSSION TITLE
LittleBigPlanet 3 and Lighthouse - Megathread

### DISCUSSION THREAD

**Discussion by m88youngling** [LBPUnion/ProjectLighthouse#452] [Ideas]
I'm creating this megathread in the hopes that we can tackle a few issues related to LittleBigPlanet 3. There are a few existing issues that I'll make a note of, 

https://github.com/LBPUnion/ProjectLighthouse/issues/270

https://github.com/LBPUnion/ProjectLighthouse/issues/202

https://github.com/LBPUnion/ProjectLighthouse/issues/453

Some additional missing features and issues, 

- [x] LittleBigPlanet 3 crashes upon connecting to Lighthouse on PS3. 
- [x] LBP3 playlists are not implemented. 
- [x] Adventures are not implemented.
- [ ] Dive In is not functional. 
- [ ] Challenges 

There are other issues that need to be addressed as well, but I can't think of them right now. 

I will start with some opening context about LBP3's compatibility issues with Lighthouse. Firstly, recently on our instance, Beacon, we implemented Cloudflare protection on the server. However, according to @Slendy , "the game's requests get rejected by Cloudflare because its SSL is so outdated". By configuring nginx a bit, we were able to get LBP3 working with http instead. However, this only got me past the announcement screen before it crashed. RPCS3 players report that the game works fine. 

This difference between PS3 and RPCS3 suggests there's something fishy going on. We need more information. Is it the client? Does the server need to provide additional information to the client to make it behave? We're not sure. @Zaprit suggested that we should use a DEX console to help us determine the cause of the crash. I figure that packet captures may also help, but I'm having trouble figuring out how to capture packets from my PS3. Connecting it to a hotspot is only possible with my Android phone, and I'm not sure how to get meaningful captures like I would with Wireshark. 

If anyone is out there with a DEX console, has packet captures to contribute, or any other information that might be helpful for resolving compatibility and feature issues mentioned anywhere above, your assistance is greatly appreciated!

**Comment by Salamni** [comment]
sup wyd wya

**Reply by W0lf4llo** [reply]
Agreed

**Comment by TorutheRedFox** [comment]
ok so, you might want to try enabling older cipher suites in Cloudflare, and as for getting LBP3 to work over HTTPS, the game also needs to be patched with the root certificate that's used to sign the gameserver's certificate (which has to be self signed as major CAs use root certs that are too big)

(see #431)

**Reply by Slendy** [reply]
As far as I know, it's not possible to enable older cipher suites in Cloudflare because they are considered insecure. You can read more about it [here](https://developers.cloudflare.com/ssl/ssl-tls/cipher-suites/), but TLDR; you can only disable certain cipher suites that are already supported by Cloudflare.

**Reply by TorutheRedFox** [reply]
huh

well that's problematic

**Comment by W0lf4llo** [comment]
From what I remember correctly jvyden’s game froze while accessing LBP3 on RCPS3. This was while he was helping me figure out why my level wouldn’t publish. The issue was that the organizatron wasn’t implemented into the servers yet. The issue has been fixed and my level has been published. So I’m sure the game still crashes but it might not be as often as the PS3

**Comment by m88youngling** [comment]
Checked a few boxes off, only remaining one is playlists.

**Reply by W0lf4llo** [reply]
Playlists are in the PR’s :)

**Comment by m88youngling** [comment]
Dive in was implemented, right? So all primary features of LBP3 (aside from more advanced community panel stuff and minor things I can't think of rn) are complete?

**Reply by W0lf4llo** [reply]
The button works but dive in itself has to be tweaked in order for it to actually work.

**Comment by m88youngling** [comment]
I added Challenges to the list, per request by penguinskillcaps.

**Comment by W0lf4llo** [comment]
Dive in is implemented and better than the original servers. It will ask to join people throughout lbp3, but if you’re hosting one it’ll instantly join that player hosting it. So there’s more people to join and kinda works like lbp2 dive in :)
